src/commonMain/kotlin/baaahs/Pinky.kt

```diff
@@ -96,22 +98,28 @@ class Pinky(
         brainsChanged = true
     }
 
-    inner class BeatProvider(val bpm: Float) {
-        var startTimeMillis = 0L
-        var beat = 0
-        var beatsPerMeasure = 4
+    inner class BeatProvider(var bpm: Float) {
```

Comment:
Assume the bpm will change every frame. How do we update this var?

I believe the mixer does a continuous sliding window sample and gives its best guess at any point. This is actually even better than relying on a single track-wide value provided by rekordbox or any other track analyzer - those tend to be wrong for the breakdowns or vocal sections of tracks, or tracks that speed up/ slow down a lot. The mixer will actually peg things more accurately in those cases.

Reply:
Yep, agreed that we should expect it to be continuously changing... whatever the source of truth for beat sync is, it'll need to update those vars somehow; ... how that happens depends on how it's implemented? You update a car by assigning a new value to it. ;-)